runtime.md

```diff
@@ -14,4 +14,40 @@ Runs a process in a container. Can be invoked several times.
 
 Not sure we need that from runc cli. Process is killed from the outside.
 
-This event needs to be captured by runc to run onstop event handlers.
+This event needs to be captured by oc to run onstop event handlers.
+
+## Hooks
+Hooks allow one to run code before/after various lifecycle events of the container.
+The state of the container is passed to the hooks over stdin, so the hooks could get the information they need to do their work.
+
+Hook paths are absolute and are executed from the host's filesystem.
+
+### Pre-start
+The pre-start hooks are called after the container process is started but before the user supplied command is executed.
+It provides an opportunity to customize the container. In Linux, for e.g., the network namespace could be configured in this hook.
```

Comment:
See @crosbymichael comment above. We aren't guaranteeing the cwd for the hook. However, it gets the container rootfs path as part of the state, so it can switch to that directory if desired.